docs/develop/python/testing-suite.mdx

```diff
@@ -36,6 +36,34 @@ Some SDKs have support or examples for popular test frameworks, runners, or libr
 
 One recommended framework for testing in Python for the Temporal SDK is [pytest](https://docs.pytest.org/), which can help with fixtures to stand up and tear down test environments, provide useful test discovery, and make it easy to write parameterized tests.
 
+### A note on pytest fixtures
+
+When using pytest fixtures to setup databases, application servers and common infrastructure it is not unusual to use the `session` scope to ensure the fixture is shared among multiple tests. However, with the test server it is important that each test is executed in isolation. Launching a test server per test introduce negligible overhead and ensures your test do not conflict with each other
+
+
+```python
+import pytest
+import temporalio.testing.WorkflowEnvironment
+
+
+# This might break because it will reuse the workflow environment 
+pytest.fixture(scope="session")
+def workflow_environment():
+  return WorkflowEnvironment.start_time_skipping()
+
+def test_1(workflow_environment):
+  pass
+
+def test_2(workflow_environment):
+  pass
+
+# Correct: if you don't specify a scope, the default scope is function,
+# so a new workflow environment will be created for each test
+@pytest.fixture 
+def workflow_environment():
+  return WorkflowEnvironment.start_time_skipping()
```

Comment:
It is a bit strange to have the primary code be the wrong code and have the same code correctly below (which you don't usually redefine the same function in the same code block). Maybe something like:

```suggestion
# Notice how this doesn't specify a "scope" like "session" scope. If you don't
# specify a scope, the default scope is function, so a new workflow
# environment will be created for each test.
@pytest.fixture
def workflow_environment():
  return WorkflowEnvironment.start_time_skipping()

def test_1(workflow_environment):
  pass

def test_2(workflow_environment):
  pass
```

Or if you must have the incorrect way be the in the initial block, can you move the correct way into its own separate block? It's a bit confusing to see the same method defined twice in the same block.